src/python/pants/backend/docker/util_rules/docker_build_context.py

```diff
@@ -50,27 +50,67 @@ class DockerBuildContextError(Exception):
 
 
 class DockerVersionContextError(ValueError):
-    pass
+    @classmethod
+    def attribute_error(
+        cls, context: DockerVersionContextValue, attribute: str
+    ) -> DockerVersionContextError:
+        msg = f"The placeholder {attribute!r} is unknown."
+        if context:
+            msg += f' Try with one of: {", ".join(context.keys())}.'
+        return cls(msg)
 
 
 class DockerVersionContextValue(FrozenDict[str, str]):
     """Dict class suitable for use as a format string context object, as it allows to use attribute
     access rather than item access."""
 
+    _attribute_error_type: ClassVar[type[DockerVersionContextError]] = DockerVersionContextError
+
+    @classmethod
+    def create(
+        cls, value: Mapping[str, str] | DockerVersionContextValue
+    ) -> DockerVersionContextValue:
+        """Create new instance of `DockerVersionContextValue` unless `value` already is an instance
+        (or subclass) of `DockerVersionContextValue`, in which case `value` is returned as-is."""
+        if isinstance(value, cls):
+            return value
+        return cls(value)
+
     def __getattr__(self, attribute: str) -> str:
         if attribute not in self:
-            raise DockerVersionContextError(
-                f"The placeholder {attribute!r} is unknown. Try with one of: "
-                f'{", ".join(self.keys())}.'
-            )
+            raise self._attribute_error_type.attribute_error(self, attribute)
         return self[attribute]
 
 
+class DockerVersionContextBuildArgError(DockerVersionContextError):
+    @classmethod
+    def attribute_error(
+        cls, context: DockerVersionContextValue, attribute: str
+    ) -> DockerVersionContextError:
+        msg = f"The build arg {attribute!r} is undefined."
+        if context:
+            msg += f' Defined build args are: {", ".join(context.keys())}.'
+        msg += (
+            "\n\nThis build arg may be defined in `pants.toml` under `[docker].build_args`, on the "
+            "command line with `--docker-build-args` or directly on the `docker_image` target "
```

Comment:
It can also be set via env var. Probably better is to simply say "with the option `[docker].build_args`". A common misconception is that some options can only be via config and others only via flag -- saying `pants.toml` like this might reinforce that.